### ISSUE TITLE
Container Hovered Style

### ISSUE THREAD

**Issue by johannesvollmer** [hecrj/iced#972]
Hi! Is it correct that the container currently does not have a hovered style? Would you accept a PR?

**Comment by 13r0ck** [comment]
You are correct, the is not a style for hovering a container. There are a few technical reasons for it
Here is Hecrj's input from the Zulip chat about events for entering a container: 
```
"
However, `on_mouse_enter` has implementation challenges. What happens if the `view` changes and a new widget ends up under the mouse cursor? The user would expect the event to trigger, but currently it won't.
"
```
Still not a no though. PRs are always welcome 😃 , just no promises that they will get merged (and will likely take a while even if they are)

**Comment by yusdacra** [comment]
Why not just use a button?

**Comment by johannesvollmer** [comment]
I'm trying to display a list of complex items. Each item is a Row. When the user hovers the row, the background color of the whole row should change. I didn't think this could be implemented with a button, especially since the row will contain buttons itself. Will it be possible to use a button for this? That would be nice

**Comment by hecrj** [comment]
I think this use case should be tackled directly by some kind of `Table` widget.

**Comment by tfachmann** [comment]
I've seen `iced` version `0.14` will include a `Table` widget. Will it be intended for such use-case?

> I'm trying to display a list of complex items. Each item is a Row.
>> Why not just use a button?

I am myself thinking about just using a button as there will be no button inside, but I am unsure whether that's the correct approach

**Comment by plorefice** [comment]
I'm facing the same exact issue and a solution that is working fairly well for me and doesn't feel hack-ish like using a button is to surround the item container with a `MouseArea` and listening for `on_enter` and `on_exit` events to set an `hovered` flag in my row item. Then I can style the container differently depending on whether it is hovered or not. Something like this:

```
            mouse_area(
                container(row)
                    .style(|theme| match msg.is_hovered {
                        true => theme::hovered_row(theme),
                        false => container::Style::default(),
                    }),
            )
            .on_enter(Message::MouseEnter(msg.id))
            .on_exit(Message::MouseLeave(msg.id))
            .into()
```
